tiledb/sm/array/array_directory.cc

```diff
@@ -324,6 +446,93 @@ tuple<Status, optional<URI>> ArrayDirectory::load_fragment_metadata_dir_uris() {
   return {Status::Ok(), std::move(fragment_meta_uri.value())};
 }
 
+tuple<
+    Status,
+    optional<std::vector<URI>>,
+    optional<std::unordered_set<std::string>>>
+ArrayDirectory::load_consolidated_commit_uris(
```

Comment:
I think this function does way more than it states. Can you please document in detail? Also does it make sense to break this function to multiple ones?